site/content/blog/2023-12-08-automated-zap-scans-for-orchard-core-apps/index.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Automated ZAP Scans for Orchard Core Apps"
+summary: >
+  If you have an app running on the ASP.NET Core web framework and CMS Orchard Core, you can now easily run ZAP scans for it.
+images:
+- https://www.zaproxy.org/blog/2023-12-08-automated-zap-scans-for-orchard-core-apps/images/security-scan-test.png
+type: post
+tags:
+- blog
+- automation
+- community
+- docker
+date: "2023-12-08"
+authors:
+- zoltan
+---
+
+ZAP is a great tool if you're doing pentesting hands-on, but for us at [Lombiq](http://lombiq.com/) it's even more valuable that it [can be thoroughly automated](/docs/automate/). This is why we've just released a tool to run ZAP security scans automatically as part of a UI test suite for Orchard Core apps.
+
+## Orchard Core and ZAP
+
+[Orchard Core](https://orchardcore.net/) is an open source ASP.NET Core web framework and Content Management System. It's a great one, as well, but due to spending the last 12 years with it, I may be a bit biased. Anyway, if you're a .NET web developer, I highly recommend you check it out so you don't need to reinvent the wheel in your own projects for things like content/user/media management, audit trail, workflows, and much more.
+
+So, where do ZAP and Orchard Core meet? Now, as part of the open source [Lombiq UI Testing Toolbox for Orchard Core](https://github.com/Lombiq/UI-Testing-Toolbox) you can run ZAP scans for Orchard Core apps, right in a UI test suite, simply by using the .NET configuration API or completely custom Automation Framework plans even.
+
+## Running ZAP from a UI Test
+
+This is what such a ZAP scan-running UI test looks like, written in C#, using the [xUnit](https://xunit.net/) testing framework:
+
+```java
```

Comment:
The code is C#, but it seems that syntax highlighting for that is not working, though [it's supported by Hugo](https://gohugo.io/content-management/syntax-highlighting/#list-of-chroma-highlighting-languages).

Reply:
That's what I tried (and `c#`) but neither resulted in the code block getting syntax highlighting (the text just stayed all white).

Reply:
Are you actually building/deploying locally?

Reply:
Yes, I've run the site, and hence why I see it not working.

Reply:
Okay was just trying to make sure it was live and not you trusting GitHub's rendering 😉